### PR TITLE
eth/server: Dumb down finding balance.

### LINKED ISSUE #2016: eth/server: Problem getting balances.
Seeing the error `2022-12-30 17:20:21.193 [ERR] MKT: (*DEXBalancer).CheckBalance: error getting account balance for %q: %v 0x946dfaB1AD7caCFeF77dE70ea68819a30acD4577 accountBalance error: contentFrom error: the method txpool_contentFrom does not exist/is not available` on master. Probably due to a recent geth version change. I do not think I noticed this in the pr, however.

### PR DESCRIPTION
**Pull request by JoeGruffins**
eth/server: Dumb down finding balance.

    eth/jwt: Add eth module check.

closes #2016

### CODE REVIEW

**Comment by JoeGruffins** [comment]
Also ok to rewind the jwt endpoint. Just offering this up as another option.

**Comment by chappjc** [comment]
Are we absolutely certain that specifying the apis via `--http.api=...` (and `--ws.api=...`) does not affect the apis provided on the `authrpc` interface as well?  The Prysm docs seem to suggest so.

**Comment by chappjc** [comment]
Seeking clarification from geth devs: https://github.com/ethereum/go-ethereum/issues/26406

**Comment by JoeGruffins** [comment]
Looks like #2023 is preferred, so closing.

File: server/asset/eth/rpcclient.go [hunk 1]
```diff
@@ -183,92 +182,14 @@ func (c *rpcclient) transaction(ctx context.Context, hash common.Hash) (tx *type
...
-	// We need to subtract and pending outgoing value, but ignore any pending
-	// incoming value since that can't be spent until mined. So we can't using
-	// PendingBalanceAt or BalanceAt by themselves.
-	// We'll iterate tx pool transactions and subtract any value and fees being
-	// sent from this account. The rpc.Client doesn't expose the
-	// txpool_contentFrom => (*TxPool).ContentFrom RPC method, for whatever
-	// reason, so we'll have to use CallContext and copy the mimic the
-	// internal RPCTransaction type.
-	var txs map[string]map[string]*RPCTransaction
-	err = c.caller.CallContext(ctx, &txs, "txpool_contentFrom", addr)
```
Comment (chappjc) [inline server/asset/eth/rpcclient.go:200]: I haven't found any publicly available (paid or not) RPC providers that have any methods from the `txpool` namespace available.  I'm thinking we can have server identify which method to take on connect via the `CheckAPIModules` function you added, however I'm not seeing `rpc_modules` supported anywhere either!  Perhaps using a third-party RPC provider with server is gonna be trickier than I'd like.

Comment (JoeGruffins) [reply server/asset/eth/rpcclient.go:200]: Yeah, when I tested the modules were only available from geth itself.

Comment (chappjc) [reply server/asset/eth/rpcclient.go:200]: Been stewing on this, and I think we should revive it with a mini project to support an RPC provider that does not provide it, but use the smart balance if it does.

The aim would then be to allow the operator to use a third party RPC provider that they trust, ideally with their account's own API key so rate limits aren't so prevalent.

This is going to be key to make testnet or just mainnet "testing" less of burden.  At present, it's ridiculously difficult, time consuming, and expensive to get geth+prysm running on either networkk.

**Pull request closed without merging**